charts/gateway-helm/templates/envoy-gateway-rbac.yaml

```diff
@@ -1,9 +1,14 @@
 {{ $watchedNamespaces := list }}
 {{ if $kube := .Values.config.envoyGateway.provider.kubernetes }}
-{{ if and $kube.watch $kube.watch.namespaces }}
+{{ if $kube.watch }}
+{{ if or (not $kube.watch.namespaces) (eq (len $kube.watch.namespaces) 0) }}
+{{ $watchedNamespaces = list }}
```

Comment:
this is also defined on line 1

Reply:
is the new commit satisfying?

Reply:
looks good to me :) 
@Xunzhuo @zirain can you help review